addons/point_of_sale/tests/test_anglo_saxon.py

```diff
@@ -155,19 +161,51 @@ def test_2041208(self):
             'amount_tax': 0,
             'amount_paid': 0,
             'amount_return': 0,
+        }
+
+        return self.PosOrder.create(pos_order_values)
+
+    def test_fifo_valuation_no_invoice(self):
+        pos_order_pos0 = self._prepare_pos_order()
+        context_make_payment = {"active_ids": [pos_order_pos0.id], "active_id": pos_order_pos0.id}
+        self.pos_make_payment_0 = self.PosMakePayment.with_context(context_make_payment).create({
+            'amount': 7 * 450.0,
         })
 
-        context_make_payment = {"active_ids": [self.pos_order_pos0.id], "active_id": self.pos_order_pos0.id}
+        # register the payment
+        context_payment = {'active_id': pos_order_pos0.id}
+        self.pos_make_payment_0.with_context(context_payment).check()
+
+        # validate the session
+        current_session_id = self.pos_config.current_session_id
+        current_session_id.action_pos_session_validate()
+
+        # check the anglo saxon move lines
+        line = pos_order_pos0.account_move.line_ids.filtered(lambda l: l.debit and l.account_id == self.category.property_account_expense_categ_id)
+        self.assertEqual(pos_order_pos0.account_move.journal_id, self.pos_config.journal_id)
+        self.assertEqual(line.debit, 27, 'As it is a fifo product, the move\'s value should be 5*5 + 2*1')
+
+    def test_fifo_valuation_with_invoice(self):
+        pos_order_pos0 = self._prepare_pos_order()
+        context_make_payment = {"active_ids": [pos_order_pos0.id], "active_id": pos_order_pos0.id}
         self.pos_make_payment_0 = self.PosMakePayment.with_context(context_make_payment).create({
             'amount': 7 * 450.0,
             'journal_id': self.cash_journal.id,
         })
 
-        # I click on the validate button to register the payment.
-        context_payment = {'active_id': self.pos_order_pos0.id}
+        # register the payment
+        context_payment = {'active_id': pos_order_pos0.id}
         self.pos_make_payment_0.with_context(context_payment).check()
+
+        # Create the customer invoice
+        pos_order_pos0.action_pos_order_invoice()
+        pos_order_pos0.account_move.post()
+
+        # validate the session
         current_session_id = self.pos_config.current_session_id
         current_session_id.action_pos_session_validate()
 
-        line = self.pos_order_pos0.account_move.line_ids.filtered(lambda l: l.debit and l.account_id == self.category.property_account_expense_categ_id)
+        # check the anglo saxon move lines
+        line = pos_order_pos0.account_move.line_ids.filtered(lambda l: l.debit and l.account_id == self.category.property_account_expense_categ_id)
+        self.assertEqual(pos_order_pos0.account_move.journal_id, self.sale_journal)
```

Comment:
the diferences between the two tests are not obvious, is is just the journal_id the difference?

Reply:
well, yes... But that difference is quite big as in this case it is the customer invoice, and in the other case it is the entry for the pos session. The lines should be in either one or the other depending on the invoicing policy for the pos.order.

Reply:
ok i finally understand the issue you're trying to fix
i'm not sure but shoudln't it be self. pos_config.invoice_journal_id ?

Reply:
then again why did it stop working in master?

Reply:
accountapocalypse, and yes but its the same thing.

Reply:
> and yes but its the same thing 

yes but it's clearer for someone reading the test and comparing with the other test

Reply:
> accountapocalypse 

was it really working before though? i quickly grepped around the v12 code and i don't see how it could use the invoice_journal_id for the anglo saxon entries .. but i don't really want to dig further so it's ok